packages/@lwc/engine-core/src/framework/rendering.ts

```diff
@@ -394,6 +395,25 @@ function patchCustomElement(
             // in fallback mode, the allocation will always set children to
             // empty and delegate the real allocation to the slot elements
             allocateChildren(n2, vm);
+
+            // Solves an edge case with slotted VFragments in native shadow mode.
```

Comment:
I couldn't find it in the documentation, but was there ever a discussion around using Comment nodes as VFragment delimiters rather than text nodes?

Comment nodes aren't slottable nodes so they shouldn't cause any observable functionality changes. Outside of slots, they should have an equal or smaller impact than text nodes.
The main downside I can think of is the DOM being visibly cluttered with comment nodes.

Reply:
From what I recall, there may have been a concern about perf as well. I.e. what is the cost of inserting comments vs text nodes?

Cluttering up the DOM is a real concern as well, though. I imagine that (and the fact that Vue does it this way) is the main reason we went with text nodes?